pydantic/_internal/_generate_schema.py

```diff
@@ -1481,6 +1481,8 @@ def _union_is_subclass_schema(self, union_type: Any) -> core_schema.CoreSchema:
     def _subclass_schema(self, type_: Any) -> core_schema.CoreSchema:
         """Generate schema for a Type, e.g. `Type[int]`."""
         type_param = self._get_first_arg_or_any(type_)
+        # Assume `type[Annotated[<typ>, ...]]` is equivalent to `type[<typ>]`:
+        type_param = _typing_extra.annotated_type(type_param) or type_param
```

Comment:
@adriangb, are you ok with this assumption?

If so, then I'm happy to move forward with this PR. Thanks @Viicos!

Reply:
Yup I think it's reasonable